### PR TITLE
Enable dynamically allocated credentials for Vault Plugin Configuration

### PR DESCRIPTION
**Pull request by lenaing**
Hello !

This PR enable use of dynamically allocated credentials in the Vault Plugin's configuration for pipelines rather than fixed credentials. Doing so will override any fixed credential ID set.

A potential usecase would be to create a credential based on external resources.
As an example, a `VaultAppRoleCredential` can be created with it's `RoleID` or `SecretID` retrieved from trusted entities. As a consequence, this enable the use of single-use and short-lived `SecretID`.

I believe others might find creative ways to use such kind of service.

Kind regards,

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,6 +125,32 @@ node {
...
+```groovy
+import hudson.util.Secret
+import com.cloudbees.plugins.credentials.CredentialsScope
+import com.datapipe.jenkins.vault.credentials.VaultTokenCredential
+
+VaultTokenCredential customCredential = new VaultTokenCredential(
+    CredentialsScope.GLOBAL,
+    'custom-credential',
+    'My Custom Credential',
+    Secret.fromString('This is my token. There are many like it, but this one is mine. My token is my best friend.')
```
Comment (Lucas-C) [inline README.md]: ❤️

**Comment by Jetersen** [comment]
link to see current commit author: https://github.com/jenkinsci/hashicorp-vault-plugin/commit/a68755e1642dbf0d34911546f28f51ded39e99ef.patch

@lenaing 
You might want to set up an email alias on your GitHub account 🥇
https://help.github.com/articles/adding-an-email-address-to-your-github-account/

Or look into fixing your git user.email config 😅  
https://help.github.com/articles/setting-your-commit-email-address-in-git
then you would need to amend your commits: https://stackoverflow.com/a/3042512
after changing author

to get proper credit for your commits 🏆

**Comment by lenaing** [comment]
Thanks, I added an alias :)

Well, the last fix I pushed was to get tests running on our GitLab-CI infrastructure with DIND. I don't understand why it fails on `windows-latest`.
It also works on a bare Windows 10...

Do you have any clue about the given error ?

**Comment by Jetersen** [comment]
Seems like a bug in their Windows Environment: https://github.com/actions/virtual-environments/issues/635

I think we can disable windows for the time being 😓

**Pull request merged**